### PR TITLE
[Codegen][GPU] Add pass to generalize named convolution ops

### PR DESCRIPTION
**Pull request by qedawkins**
After tiling the filter dimensions to 1, the simplest way to both handle residual unit dimensions and vectorize the convolution is to generalize it. This adds a pass for generalizing all convolution ops with this intended use.

### CODE REVIEW

**Review by kuhar — APPROVED** [review]
LGTM but maybe wait for another approval before landing

**Review by antiagainst — APPROVED** [review]

**Pull request merged**
